resources/aws/gateway.go

```diff
@@ -103,15 +88,18 @@ func (g *Gateway) CreateOrFail() error {
 }
 
 func (g *Gateway) Delete() error {
-	gateway, err := g.findExisting()
+	gateways, err := g.list()
 	if err != nil {
 		return microerror.MaskAny(err)
 	}
 
-	if _, err := g.Clients.EC2.DeleteInternetGateway(&ec2.DeleteInternetGatewayInput{
-		InternetGatewayId: gateway.InternetGatewayId,
-	}); err != nil {
-		return microerror.MaskAny(err)
+	for _, gateway := range gateways {
+		_, err := g.Clients.EC2.DeleteInternetGateway(&ec2.DeleteInternetGatewayInput{
+			InternetGatewayId: gateway.InternetGatewayId,
+		})
+		if err != nil {
+			return microerror.MaskAny(err)
+		}
 	}
```

Comment:
I agree with  @nhlfr and actually start to like previous design more. I didn't realize that `Delete` may return these errors. I'm closing the PR.

One thing, which is still disturbing though is that `findExisting` methods may possibly return indeterministic resource if filters are incorrect, or something else creates a resource that matches filters.
Perhaps there should be a stronger assertion like `len(gateways.InternetGateways) == 1`? I guess in that case  human intervention is required.